arex-web-core/src/main/java/com/arextest/web/core/business/config/application/ApplicationOperationConfigurableHandler.java

```diff
@@ -33,7 +46,43 @@ public boolean insert(ApplicationOperationConfiguration configuration) {
         return super.insert(configuration);
     }
 
+    @Override
+    public boolean update(ApplicationOperationConfiguration configuration) {
+        // cover contract manually
+        if (configuration.getOperationResponse() != null) {
+            String contract = SchemaUtils.mergeJson(null, configuration.getOperationResponse());
+            if (contract != null) {
+                AppContractDto appContractDto = new AppContractDto();
+                appContractDto.setOperationId(configuration.getId());
+                appContractDto.setIsEntryPoint(true);
+                appContractDto.setOperationName(configuration.getOperationName());
+                appContractDto.setContract(contract);
+                appContractRepository.upsertAppContractList(Collections.singletonList(appContractDto));
+            }
+        }
+        return super.update(configuration);
+    }
+
     public ApplicationOperationConfiguration useResultByOperationId(String operationId) {
-        return applicationOperationConfigurationRepository.listByOperationId(operationId);
+        ApplicationOperationConfiguration result =
+                applicationOperationConfigurationRepository.listByOperationId(operationId);
+        List<AppContractDto> appContractDtoList = appContractRepository.queryAppContractList(operationId);
+
+        List<Dependency> dependencyList = new ArrayList<>();
+
+        for (AppContractDto appContractDto : appContractDtoList) {
+            if (result.getOperationTypes().contains(appContractDto.getOperationType())) {
+                // only match once
+                result.setResponseContract(appContractDto.getContract());
```

Comment:
Use a unified interface to query interfaces and dependency contracts. No special use of responseContract to handle interface contracts

Reply:
fixed